src/hooks/use-dao-details.ts

```diff
@@ -1,307 +1,317 @@
 import { useState, useEffect } from "react";
 import { supabase } from "@/utils/supabase/client";
 import { Extension, DAO, Holder, Token } from "@/types/supabase";
+import FaktorySDK from "@faktoryfun/core-sdk";
 
 interface MarketStats {
-  price: number;
-  marketCap: number;
-  treasuryBalance: number;
-  holderCount: number;
+    price: number;
+    marketCap: number;
+    treasuryBalance: number;
+    holderCount: number;
 }
 
 interface TreasuryToken {
-  type: 'FT' | 'NFT';
-  name: string;
-  symbol: string;
-  amount: number;
-  value: number;
+    type: 'FT' | 'NFT';
+    name: string;
+    symbol: string;
+    amount: number;
+    value: number;
 }
 
 interface TokenBalance {
-  balance: string;
-  total_sent: string;
-  total_received: string;
+    balance: string;
+    total_sent: string;
+    total_received: string;
 }
 
 interface HiroBalanceResponse {
-  stx: TokenBalance;
-  fungible_tokens: {
-    [key: string]: TokenBalance;
-  };
-  non_fungible_tokens: {
-    [key: string]: {
-      count: number;
-      total_sent: number;
-      total_received: number;
+    stx: TokenBalance;
+    fungible_tokens: {
+        [key: string]: TokenBalance;
+    };
+    non_fungible_tokens: {
+        [key: string]: {
+            count: number;
+            total_sent: number;
+            total_received: number;
+        };
     };
-  };
 }
 
 interface HiroHolderResponse {
-  total_supply: string;
-  limit: number;
-  offset: number;
-  total: number;
-  results: {
-    address: string;
-    balance: string;
-  }[];
+    total_supply: string;
+    limit: number;
+    offset: number;
+    total: number;
+    results: {
+        address: string;
+        balance: string;
+    }[];
 }
 
-// Cache for storing fetched data
 const daoCache = new Map<string, {
-  data: {
-    dao: DAO | null;
-    daoExtensions: Extension[] | null;
-    holders: Holder[];
-    tokenSymbol: string;
-    token: Token;
-    marketStats: MarketStats;
-    treasuryTokens: TreasuryToken[];
-  };
-  timestamp: number;
+    data: {
+        dao: DAO | null;
+        daoExtensions: Extension[] | null;
+        holders: Holder[];
+        tokenSymbol: string;
+        token: Token;
+        marketStats: MarketStats;
+        treasuryTokens: TreasuryToken[];
+    };
+    timestamp: number;
 }>();
 
 const CACHE_DURATION = 5 * 60 * 1000; // 5 minutes in milliseconds
 
+const sdkMainnet = new FaktorySDK({
```

Comment:
Shouldn't this use the method in `src/lib/faktory-fun.ts`?

Also we should be able to provide mainnet or testnet from `NEXT_PUBLIC_STACKS_NETWORK`